scripts/cicd/verify-gather-logs-operations.py

```diff
@@ -39,6 +39,15 @@
     'starter-us-west-2'
 ]
 
+# This regex matches a valid hostname: an optional set of DNS labels each
+# ending with dot, followed by a final DNS label. Each DNS label is 1 to
+# 63 digits, letters or '-', but '-' can't be the first or last character.
+HOSTNAME_RE = re.compile(
```

Comment:
Right: the regex would match IPs too - it's not strictly checking for valid DNS labels... that was the initial intention but I thought that developers might prefer the ability to specify nodes in different ways depending on how they obtain the reference to the node they are interested in (e.g. ansible output might be pointing to an inventory name instead of a hostname). This way they can use node name, inventory name, internal IP or external IP... I will adjust the comment to reflect this, if that's ok.

A verification that each of the provided node names actually refers to a real existing node and that the node belongs to the specified cluster requires a check of the inventory - this is done in the gather logs script itself (see the referenced PR). The validation here is only a sanity check to verify that it matches "something that could be in the host inventory".

The (?!-) and (?<!-) check that the next and previous character are not "-" respectively, without actually "consuming" a character match, i.e. lookahead/lookbehind checks This prevents node names like -this.one or that-.other. It's probably not necessary (argparse should complain with the former, inventory lookup would fail on the latter) so I could just remove them if they confuse more than help...